applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_embedded_solver.py

```diff
@@ -118,89 +106,94 @@ def ImportModelPart(self):
-                                                                   self.time_scheme,
-                                                                   self.trilinos_linear_solver,
-                                                                   self.conv_criteria,
-                                                                   self.builder_and_solver,
-                                                                   self.settings["maximum_iterations"].GetInt(),
-                                                                   self.settings["compute_reactions"].GetBool(),
-                                                                   self.settings["reform_dofs_at_each_step"].GetBool(),
-                                                                   self.settings["move_mesh_flag"].GetBool())
-
-        (self.solver).SetEchoLevel(self.settings["echo_level"].GetInt())
-        (self.solver).Initialize()
-
-        # Set the distance modification process
-        self.__GetDistanceModificationProcess().ExecuteInitialize()
-
-        # For the primitive Ausas formulation, set the find nodal neighbours process
-        # Recall that the Ausas condition requires the nodal neighbouts.
-        if (self.settings["formulation"]["element_type"].GetString() == "embedded_ausas_navier_stokes"):
-            number_of_avg_elems = 10
-            number_of_avg_nodes = 10
-            self.find_nodal_neighbours_process = KratosMultiphysics.FindNodalNeighboursProcess(self.GetComputingModelPart(),
-                                                                                               number_of_avg_elems,
-                                                                                               number_of_avg_nodes)
-
-        KratosMultiphysics.Logger.PrintInfo("NavierStokesMPIEmbeddedMonolithicSolver","Solver initialization finished.")
\ No newline at end of file
+            builder_and_solver = KratosTrilinos.TrilinosBlockBuilderAndSolver(
+                epetra_communicator,
+                guess_row_size,
+                trilinos_linear_solver)
+        return builder_and_solver
+
+    def _CreateSolutionStrategy(self):
+        computing_model_part = self.GetComputingModelPart()
+        time_scheme = self._GetScheme()
+        linear_solver = self._GetLinearSolver()
+        convergence_criterion = self.GetConvergenceCriterion()
+        builder_and_solver = self._GetBuilderAndSolver()
+        return KratosTrilinos.TrilinosNewtonRaphsonStrategy(
+            computing_model_part,
+            time_scheme,
+            linear_solver,
+            convergence_criterion,
+            builder_and_solver,
+            self.settings["maximum_iterations"].GetInt(),
+            self.settings["compute_reactions"].GetBool(),
+            self.settings["reform_dofs_at_each_step"].GetBool(),
+            self.settings["move_mesh_flag"].GetBool())
+
+    @classmethod
+    def _FmAleIsActive(self):
```

Comment:
this cannot be a classmethod!

Reply:
mmm... Otherwise a warning is thrown. As there is no call to `self` I thought we need the `@classmethod` specifier.

Reply:
well there is no necessity yet, but there will be (see baseclass)
btw why is this even  special in Trilinos?